test/zipkin/util.go

```diff
@@ -48,13 +48,21 @@ const (
 	ZipkinNamespace = "istio-system"
 )
 
-var zipkinPortForwardPID int
+var (
+	zipkinPortForwardPID int
+	// ZipkinTracingEnabled global variable indicating if zipkin is enabled.
+	ZipkinTracingEnabled = false
+)
 
 // SetupZipkinTracing sets up zipkin tracing which involves:
 // 1. Setting up port-forwarding from localhost to zipkin pod on the cluster
 //    (pid of the process doing Port-Forward is stored in a global variable).
 // 2. Enable AlwaysSample config for tracing.
 func SetupZipkinTracing(kubeClientset *kubernetes.Clientset) error {
+	if ZipkinTracingEnabled {
```

Comment:
This is racy and not safe.
I highly recommend to switch to `sync.Once` to do initialization like this.

Reply:
Before I make the change, wouldn't protection through mutex be more suitable here.

The intended use here was to have this method be called logical by test.setup and CleanupZipkinTracing be called by test.Cleanup, hence synchronization protection was not added.

Reply:
So is this per-test or per test-suite?
If first, then mutex, if second then Once.

Reply:
It would be per-test. I have changed to using RWMutex.